src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -2612,3 +2662,64 @@ citus_internal_delete_shard_metadata(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+
+/*
+ * citus_internal_update_relation_colocation is an internal UDF to
+ * delete a row in pg_dist_shard and corresponding placement rows
+ * from pg_dist_shard_placement.
+ */
+Datum
+citus_internal_update_relation_colocation(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	uint32 tagetColocationId = PG_GETARG_UINT32(1);
+
+	EnsureTableOwner(relationId);
+
+	if (!ShouldSkipMetadataChecks())
+	{
+		/* this UDF is not allowed allowed for executing as a separate command */
+		EnsureCoordinatorInitiatedOperation();
+
+		/* ensure that the table is in pg_dist_partition */
+		char partitionMethod = PartitionMethodViaCatalog(relationId);
+		if (partitionMethod == DISTRIBUTE_BY_INVALID)
+		{
+			/* connection from the coordinator operating on a shard */
+			ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+							errmsg("The relation \"%s\" does not have a valid "
+								   "entry in pg_dist_partition.",
+								   get_rel_name(relationId))));
+		}
+		else if (partitionMethod != DISTRIBUTE_BY_HASH)
+		{
+			/* connection from the coordinator operating on a shard */
+			ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+							errmsg("Updating colocation ids are only allowed for hash "
+								   "distributed tables: %c", partitionMethod)));
+		}
```

Comment:
These two checks don't have coverage

Reply:
manually modified metatada in a tx block to trigger